dashboard/app/models/user.rb

```diff
@@ -888,6 +891,28 @@ def in_progress_and_completed_scripts
     end
   end
 
+  # This method is meant to return a bunch of data about a user's recent courses.
+  # Right now, it just looks at recent scripts, but we expect it to change in the
+  # very near future
+  # TODO: Once this data is more real, we should be writing tests for it.
+  def recent_courses
+    in_progress_and_completed_scripts.map do |script|
```

Comment:
nit: Make this `do |user_script|`, as `in_progress_and_completed_scripts` returns a collection of `user_scripts` (not `scripts`). Doing so explains your hitting the cache for the name rather than (the expected) `script.name`.